### PR TITLE
User metrics with optional platform specific db persistence

### PR DESCRIPTION
**Pull request by varoonp123**
Like #58 but with a massaged commit history. 

- Hoist the arcade-specific features (serial ports) to another crate to assist with conditional compilation. Hoist whatever bits were needed into the `thetawave_interface` crate to be used as something like a "service registry" with schemas+typedefs.
- Implement a plugin to keep track of 4 caches of user metrics+mob kills for the current game and all completed games.
- Implement a plugin to periodically persist the completed games caches to a local sqlite. 
- Show the metrics on the Game over, main menu, and victory screens.

### CODE REVIEW

File: crates/thetawave_arcade/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "thetawave_arcade"
+version = "0.1.0"
+edition = "2021"
```
Comment (varoonp123) [inline crates/thetawave_arcade/Cargo.toml:4]: Use consistent package version for all crates in the workspace.

File: crates/thetawave_interface/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+serde = {workspace = true}
+strum = {workspace = true}
+strum_macros = {workspace = true}
+derive_more = {workspace = true}
+bevy_ecs_macros = "0.11.0"
+bevy_ecs = "0.11.0"
```
Comment (varoonp123) [inline crates/thetawave_interface/Cargo.toml:14]: Use consistent bevy versions across all deps. Ideally we only update 1 place when they release a new version.

File: crates/thetawave_storage/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+bevy = {workspace = true}
+thiserror = {workspace = true}
+derive_more = {workspace = true}
+rusqlite = { version = "0.29.0", features = ["bundled"] }
+dirs = "5.0.1"
```
Comment (varoonp123) [inline crates/thetawave_storage/Cargo.toml:13]: I think `directories` is slightly higher leveled. Use that instead.

**Review by cdsupina — APPROVED** [review]
Good one 🤣

**Pull request merged**